EmmyLua/SharedXML/LinkUtil.lua

```diff
@@ -0,0 +1,35 @@
+---@param linkString string
+---@return boolean containsHyperlink
+---@return string|nil preString
```

Comment:
We should imho try to use `?` instead of `|nil` unless it's a `@field`
```
---@return string? preString`
```

Reply:
I don't think it's a good idea. For example, I primarily associate a question mark with optional values, not nilable ones. Anyway, it's up to you

Reply:
You do have a point there, the question mark makes more sense for optional values instead of nilable.

For wowpedia docs we did switch to the question mark even for nilable values. One of the reasons was that Townlong Yak formats this the same way.
https://www.townlong-yak.com/framexml/ptr/Blizzard_APIDocumentation#C_BattleNet.GetGameAccountInfoByID